nativeruntime/cpp/androidfw/CursorWindow.cpp

```diff
@@ -62,9 +81,18 @@ status_t CursorWindow::create(const String8& name, size_t size,
     if (result < 0) {
       ALOGE("CursorWindow: ashmem_set_prot_region() failed: errno=%d", errno);
     } else {
+      // void* data = ::mmap(nullptr, size, PROT_READ | PROT_WRITE, MAP_SHARED,
+      //                     ashmemFd, 0);
+      // if (data == MAP_FAILED) {
+      #if defined(_WIN32)
```

Comment:
What about just having separate functions, such as:
```cpp

#ifdef _WIN32

status_t CursorWindow::create(const String8& name, size_t size,
                              CursorWindow** outCursorWindow) {

  String8 ashmemName("CursorWindow: ");
  ashmemName.append(name);
  status_t result;

  
  void* data = malloc(size);
  CursorWindow* window =
    new CursorWindow(name, 0 /* ashmemFd */, data, size, false /*readOnly*/);
  result = window->clear();
  if (!result) {
    LOG_WINDOW(
      "Created new CursorWindow: freeOffset=%d, "
      "numRows=%d, numColumns=%d, mSize=%zu, mData=%p",
      window->mHeader->freeOffset, window->mHeader->numRows,
      window->mHeader->numColumns, window->mSize, window->mData);
    *outCursorWindow = window;
    return OK;
  }
  delete window;
  free(data);
  return result;
}

#else

// the original function

status_t CursorWindow::create(const String8& name, size_t size,
                              CursorWindow** outCursorWindow) {
...
}

#endif

Then you wouldn't need to define any extra constants or create a temp file in Windows.

Reply:
Wow. That's better. I will try it with new commit. Thanks.

Reply:
Add similar content at new commit.